sdk/core/src/factories/legacyWalletFactories.ts

```diff
@@ -0,0 +1,317 @@
+import { EventComponent } from '@affinidi/affinity-metrics-lib'
+
+import {
+  LegacyNetworkMemberWithFactories,
+  UniversalDerivedType,
+} from '../CommonNetworkMember/LegacyNetworkMemberWithFactories'
+import { Util } from '../CommonNetworkMember/Util'
+import { CognitoUserTokens, KeyParams, MessageParameters, SdkOptions } from '../dto/shared.dto'
+import { getOptionsFromEnvironment, ParsedOptions } from '../shared/getOptionsFromEnvironment'
+import { AffinidiCommonConstructor, IPlatformEncryptionTools } from '../shared/interfaces'
+
+const createWallet = (
+  platformEncryptionTools: IPlatformEncryptionTools,
+  affinidiCommon: AffinidiCommonConstructor | null,
+  component: EventComponent,
+) => {
+  class Wallet extends LegacyNetworkMemberWithFactories {
+    constructor(
+      password: string,
+      encryptedSeed: string,
+      options: ParsedOptions,
+      cognitoUserTokens?: CognitoUserTokens,
+    ) {
+      super(password, encryptedSeed, platformEncryptionTools, affinidiCommon, options, component, cognitoUserTokens)
+    }
+  }
+
+  return Wallet as UniversalDerivedType
+}
+
+export const createLegacyWalletFactories = (
```

Comment:
Where are the factories? What are these functions do, exactly? They look a bit confusing to me without context

Reply:
Originally, SDK exported a class with a bunch of static methods returning instances of that class:
```
class CommonNetworkMember {
  constructor(platformEncryptionTools, ...otherParams) {}

  static fromLoginAndPassword(login, password) { return new this(...otherParams) }
}

// in react-native
export class AffinidiWalletReactNative extends CommonNetworkMember {
  constructor(...otherParams) {
    super(reactNativePlatformTools, ...otherParams)
  }
}

// usage
const wallet = AffinidiWalletReactNative.fromLoginAndPassword(...) // here, AffinidiWalletReactNative is a class, and wallet is an instance of that class
```

Now instead of exporting a class, we have
```
const createLegacyWalletFactories = (platformEncryptionTools) => ({
  fromLoginAndPassword: (login, password) => //returns CommonNetworkMember
})

const createV6Factories = (platformEncryptionTools) => ({
  logInWithPassword: (login, password) => //returns CommonNetworkMember
})

const createFactories = // combines legacy and v6 factories

// in react-native
export const AffinidiWalletReactNative = createFactories(reactNativePlatformTools)

// usage
const legacyWallet = AffinidiWalletReactNative.fromLoginAndPassword(...) // here, AffinidiWalletReactNative is just an object, fromLoginAndPassword is its field, and legacyWallet is an instance of internal "legacy wallet" class defined in core sdk only

const walletV6 = AffinidiWalletReactNative.logInWithPassword(...) // walletV6 is an instance of internal "wallet with user management" class defined in core SDK only
```

This allowed me to split CommonNetworkMember.ts into several smaller files, and to have three different leaf classes (legacy, with cognito, without cognito) each with its own set of methods and with its own set of constructor parameters.
So that you can do `legacyWallet.signOut(options)` or `walletV6.logOut()` but not vice versa; and you cannot call `signOut` or `logOut` on an object returned by `AffinidiWalletReactNative.createFromEncryptedSeedAndPassword`.